frontend/src/app/directives/ellipsis.directive.ts

```diff
@@ -0,0 +1,46 @@
+import { AfterViewInit, Directive, ElementRef, HostListener, Inject, PLATFORM_ID } from '@angular/core';
+import { isPlatformBrowser } from '@angular/common';
+
+@Directive({
+  selector: '[ovideEllipsis]'
+})
+export class EllipsisDirective implements AfterViewInit {
+
+  private get el(): HTMLElement {
+    return this.elementRef.nativeElement;
+  }
+
+  private innerText: string;
+
+  constructor(
+    private readonly elementRef: ElementRef,
+    @Inject(PLATFORM_ID) private readonly platformId
+  ) { }
+
+  public ngAfterViewInit(): void {
+    this.truncate();
+  }
+
+  @HostListener('window:resize')
+  private onWindowResize() {
+    this.truncate();
+  }
+
+  private truncate(): void {
+    if (!isPlatformBrowser(this.platformId)) {
+      return;
+    }
+
+    if (this.innerText === undefined) {
+      this.innerText = this.el.innerText.trim();
```

Comment:
Do not use nativeElement directly, instead use [Renderer2](https://angular.io/api/core/Renderer2)